python/ray/train/_internal/backend_executor.py

```diff
@@ -424,6 +425,9 @@ def initialize_session(
             node_rank_map,
         ) = self._create_rank_world_size_mappings()
 
+        tune_session: _TrainSession = get_session()
+        assert tune_session, "`start_training` should only be called from within Tune"
```

Comment:
Rllib's `LearnerGroup` uses `BackendExecutor`, and they may not be inside a Tune session (??). But they never call this code and  only use it to start and stop a `WorkerGroup`. Maybe they should just use the `WorkerGroup` abstraction directly 😅